sdk/keyvault/azure-security-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/SecretAsyncClient.java

```diff
@@ -215,7 +215,7 @@ public Mono<Response<KeyVaultSecret>> setSecretWithResponse(KeyVaultSecret secre
      * @param name The name of the secret.
      * @return A {@link Mono} containing the requested {@link KeyVaultSecret secret}.
      * @throws ResourceNotFoundException when a secret with {@code name} doesn't exist in the key vault.
-     * @throws HttpResponseException if {@code name} is empty string.
+     * @throws IllegalArgumentException If {@code name} is either {@code null} or empty.
```

Comment:
We should still document that `HttpResponseException` can be thrown for other reasons - 4xx and 5xx errors.